### ISSUE TITLE
GO analysis and plotting scripts 

### ISSUE THREAD

**Issue by hputnam** [hputnam/Becker_E5#15]
@daniellembecker Can you please link here to the topGO analysis script and figure plotting? I would like to run through the analysis and how the bubble size is generate in the figure from the DEGs.

**Comment by daniellembecker** [comment]
@hputnam here is [the script,](https://github.com/hputnam/Becker_E5/blob/master/RAnalysis/Scripts/DEG_topGO_Enrichment_Analysis.Rmd) the bubble size is generated by taking the numDEI (significant) genes which are the number of genes in our subset (DEGs) that are annotated with the specific GO term divided by the numInCat (annotated): the total number of genes in my background gene set that are annotated with the specific GO term multiplied by 100: 

```
#Prepare data for plotting by grouping by GO.ID, direction, ParentTerm, and Term
plot_data_BP <- filtered_direction_go_results_BP %>%
  group_by(GO.ID, direction, ParentTerm, Term) %>%
  summarise(numInCat = sum(numInCat), numDEI = sum(numDEI)) %>%
  ungroup() %>%
  mutate(percentDEGs = (numDEI / numInCat) * 100)
```

**Comment by hputnam** [comment]
I am looking at your script for the GO results and I have two major areas of concern: 
1) The completion of the up and down regulated separately and then merging the results.... This is not an apples to apples comparison.
2) Using multiple incompatible approaches for data reduction. I am not following the logic of using topGO, which is based on GO graph topology, followed by a reduction based on semantic similarity, removing terms subjectively, and then combining up and down results quantitatively, and then using a bubble size for a number of genes which is more than 10-fold higher than the number of DEGS you started with.

**Comment by daniellembecker** [comment]
Just to explain the order of your number 2 comment, semantic similarity with rrvgo was used to reduce the terms for visualization purposes and show the hierarchical connections between the terms for the dendrogram with assigned parent terms to explain the higher level connections of the more targeted and specific terms. We had previously spoke about ways to reduce contents in a very large figure that would normally contain ~ 513 terms for just the DEG figure alone, for example, you suggested following [what you did for Federicas paper](https://www.sciencedirect.com/science/article/pii/S1047847723000990):where some terms were removed if they were less informative and set a cut-off for how many DEGs should be in a category like: "before final plotting by only keeping terms with at least 2 DEGs per category (numDEInCat) and by excluding broader terms such as “modulation process of other organism”, “membrane”, “component of membrane” that may be less informative." So I did this just for the visualization, while still reporting the full go enrichment in tables and throughout the manuscript. Also, the bubble size for number of genes is not showing the DEG number of genes. The 1000-3000 range of genes are the annotated genes that are the total number of genes in my background gene set that are annotated with the specific GO term, meaning any gene that could be in the background and possibly associated with the GO term are included in that number of genes, I summarized them per GO.ID, direction, and Term for the visualization so they are pretty high but still real data. In regards to the up and down merging I will have to look into that as the scripts I have been following run the up and down separately or modules separately then combine for the visualization, they are still processed and quantitatively tested separately.  Happy to chat about this more but I have been following the vignettes for these approaches and looking at previous literature and code to try and make sure I am approaching it in the correct way so just wanted to explain what I have seen and why I made these decisions.

**Comment by hputnam** [comment]
Yes, all of those things have logic and rationale independently, but I am talking about them in combination.  For example, I do not thing the theory of graph topology and semantic similarity can or should be used together.   Look again at [Alexa et al](https://academic.oup.com/bioinformatics/article/22/13/1600/193669) as well as the version of the analysis I will send you and push, specifically the graphics of the enriched graph topology.

**Comment by hputnam** [comment]
What I think makes sense for topgo is to report the top results in a table similar to this paper: https://www.pnas.org/doi/full/10.1073/pnas.2015737117#sec-3

Then to compile a list of your genes of interest from your a priori hypotheses e.g., under the categories of genes that would be responding to nitrogen and changes in symbiotic function from prior papers.
Nitrogen Metabolism Genes
Symbiosis Transport Genes 
Carbon Metabolism Genes

Then use this in topgo Predefined list of interesting genes (section 4.4 topgo)

In this way you have an unbiased approach from the DEGs as well as your targeted hypothesis approach.
